tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -301,6 +301,41 @@ llvm::Type* GetIndexTypeForKernel(const HloInstruction* hlo, int64 launch_size,
   return b->getInt32Ty();
 }
 
+// Check whether all the shapes of the slice input operands are the same.
```

Comment:
Could you clarify the contract here? It's not just "checking", since we are also returning the shape.

Something like "Returns FailedPrecondition if input shapes differ, otherwise, returns the shape of ..."

Reply:
Revised the name and added some more comments.